ray-operator/controllers/ray/rayservice_controller_test.go

```diff
@@ -562,57 +561,68 @@ var _ = Context("Inside the default namespace", func() {
 				time.Second*5, time.Millisecond*500).Should(Equal(initialClusterName), "My current RayCluster name  = %v", myRayService.Status.ActiveServiceStatus.RayClusterName)
 
 			// The cluster should switch once the deployments are finished updating
-			fakeRayDashboardClient.SetServeStatus(generateServeStatus(metav1.NewTime(time.Now().Add(time.Duration(-5)*time.Minute)), "HEALTHY"))
+			fakeRayDashboardClient.SetSingleApplicationStatus(generateServeStatus("HEALTHY"))
 			updateHeadPodToRunningAndReady(ctx, pendingRayClusterName)
 
 			Eventually(
 				getRayClusterNameFunc(ctx, myRayService),
 				time.Second*60, time.Millisecond*500).Should(Equal(pendingRayClusterName), "My current RayCluster name  = %v", myRayService.Status.ActiveServiceStatus.RayClusterName)
 		})
+
+		It("should reconcile status correctly when multi-app config type is used.", func() {
+			ctMultiApp := utils.MULTI_APP
+			serveConfigTypeForTesting = &ctMultiApp
+
+			// Set multi-application status to healthy.
+			healthyStatus := generateServeStatus("HEALTHY")
+			fakeRayDashboardClient.SetMultiApplicationStatuses(map[string]*utils.ServeApplicationStatus{"default": &healthyStatus})
+			// Set single-application status to unhealthy.
+			unhealthyStatus := generateServeStatus("UNHEALTHY")
+			fakeRayDashboardClient.SetSingleApplicationStatus(unhealthyStatus)
+
+			// The status should remain healthy
+			Consistently(
+				checkServiceHealth(ctx, myRayService),
+				time.Second*5, time.Millisecond*500).Should(BeTrue(), "myRayService status = %v", myRayService.Status)
+
+			// reset states to healthy for following test cases
+			serveConfigTypeForTesting = nil
+			fakeRayDashboardClient.SetSingleApplicationStatus(healthyStatus)
+		})
 	})
 })
 
 func prepareFakeRayDashboardClient() utils.FakeRayDashboardClient {
 	client := utils.FakeRayDashboardClient{}
 
-	client.SetServeStatus(generateServeStatus(metav1.Now(), "HEALTHY"))
+	client.SetSingleApplicationStatus(generateServeStatus("HEALTHY"))
 
 	return client
 }
 
-func generateServeStatus(time metav1.Time, status string) utils.ServeDeploymentStatuses {
-	serveStatuses := utils.ServeDeploymentStatuses{
-		ApplicationStatus: rayv1alpha1.AppStatus{
-			Status:               "RUNNING",
-			LastUpdateTime:       &time,
-			HealthLastUpdateTime: &time,
-		},
-		DeploymentStatuses: []rayv1alpha1.ServeDeploymentStatus{
-			{
-				Name:                 "shallow",
-				Status:               status,
-				Message:              "",
-				LastUpdateTime:       &time,
-				HealthLastUpdateTime: &time,
+func generateServeStatus(status string) utils.ServeApplicationStatus {
```

Comment:
`DeploymentStatusEnum`? Without this, some invalid values may be passed into this function, which may alter the code path.

Reply:
I can improve this in a follow up PR - right now `DeploymentStatusEnum` is not a type.

Reply:
This one?

https://github.com/ray-project/kuberay/blob/35fe6f93129f3458e71a1481dd93d67b376ba1c0/ray-operator/apis/ray/v1alpha1/rayservice_types.go#L38-L46

Reply:
Yeah I tried using this as a type in `generateServeStatus`, it doesn't work. I think you may need to declare `type DeploymentStatusEnum string` or something.